scanpy/plotting/_tools/scatterplots.py

```diff
@@ -725,7 +725,7 @@ def _get_color_values(adata, value_to_plot, groups=None, palette=None, use_raw=F
             color_vector = adata[:, value_to_plot].layers[layer]
         elif use_raw and value_to_plot in adata.var_names:
             color_vector = adata.raw[:, value_to_plot].X
```

Comment:
are you sure that the `.raw` shouldn’t be inserted in the test for this branch instead?

in my fix I do that and a test fails. was the test written with the wrong assumption?

Reply:
That's probably right. I just made a change so the bug would stop happening and I could work on other stuff. We can go with yours.